doc/sphinx/language/extensions/compil-steps.rst

```diff
@@ -0,0 +1,80 @@
+.. _compil-steps:
+
+======================
+Main Compilation Steps
+======================
+
+It is in theory possible to write down every Gallina term explicitly
+as described in the :ref:`Core Language <core-language>` part of this
+manual, in the way they are understood by the kernel, for instance
+:n:`Nat.add (S O) (Nat.mul (S (S O)) (S (S O)))`. However, this would
+be very tedious and error-prone and takes us away from our usual
+mathematical practice. To circumvent this, Rocq offers multiple
+preprocessing mechanisms to help fill the gap between what the users
+would like to input to the system and the fully formal core language
+expected by the kernel.
+
+For instance, the notation mechanisms reflects the eponym mathematical
+practice and allows to write :n:`1 + 2 * 2` instead of the above
+term. Those mechanisms range from simple :ref:`Abbreviations` to full
+fledged :ref:`Notations` with user defined :ref:`syntaxes
+<ReservingNotations>`. Multiple interpretations can be given to the
+same syntax in different contexts thanks to the :ref:`scope
+<Scopes>` mechanism. For instance :n:`(1 + 2 * 2)%N` could denote the
+above natural number expression while :n:`(1 + 2 * 2)%Z` could denote
+a similar expression with integers.
```

Comment:
```suggestion
<Scopes>` mechanism. For instance :n:`(1 + 2 * 2)%N` is
a natural number expression while :n:`(1 + 2 * 2)%Z` is
an expression with integers.
```

simpler language.

Reply:
This is also changing the meaning. I tried to preserve it.